### PR TITLE
[DEV-4775] New Endpoint for Award Type Groups

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**Description:**
Adds a new end point returning a JSON object that describes the grouping of award types.

**Technical details:**
- endpoint is `v2/references/award_types`
- API Contract ✅ 
- No tests b/c no models were created? Looked in tests dir for `references` app and could not find an example of a non-model being tested. Please correct me here if I need a test.

**Requirements for PR merge:**

1. `N/A` Unit & integration tests updated
2. [x] API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. `N/A` Matview impact assessment completed
5. `N/A` Frontend impact assessment completed
6. `N/A` Data validation completed
7. `N/A` Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-4775](https://federal-spending-transparency.atlassian.net/browse/DEV-4775):
    - [x] Link to this Pull-Request
    - `N/A`Performance evaluation of affected (API | Script | Download)
    - `N/A` Before / After data comparison

**Area for explaining above N/A when needed:**
```
New endpoint with only a view that returns pre-existing data in the `api_helper.py`
```

### CODE REVIEW

File: usaspending_api/references/v2/views/award_types.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class AwardTypeGroups(APIView):
+    """
+    This route returns a JSON object describing the award type groupings.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/references/award_types.md"
+
+    def get(self, request, format=None):
+        return Response({
```
Comment (tony-sappe) [inline usaspending_api/references/v2/views/award_types.py:21]: Wouldn't it be simpler to construct the response by combining the `assistance_type_mapping` and `procurement_type_mapping` dicts?

File: usaspending_api/api_contracts/contracts/v2/references/award_types.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+This endpoint returns a JSON object representing the grouping of award types.
+
+## GET    
++ Response 200 (application/json)
+    + Attributes (object)
+        + `results` (required, AwardTypes, fixed-type)
+    + Body
+            {
+                "results": {
```
Comment (tony-sappe) [inline usaspending_api/api_contracts/contracts/v2/references/award_types.md:14]: Is it beneficial for clients to have the data nested under `results`? It seems extraneous

File: usaspending_api/references/v2/views/award_types.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    """
+    This route returns a JSON object describing the award type groupings.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/references/award_types.md"
+
+    def get(self, request, format=None):
+        return Response(
+            {
+                "results": {
```
Comment (BrianZito) [inline usaspending_api/references/v2/views/award_types.py:23]: I agree with Tony's old comment: This isn't a search, so I don't think it's productive to put it in a results array. The ticket mentions an `award_type_groups` key serving this purpose.

Comment (tony-sappe) [reply usaspending_api/references/v2/views/award_types.py:23]: Is it necessary to nest the award categories under a key? Each category could be a top-level key

File: usaspending_api/references/v2/views/award_types.py (hunk 1 above)
Comment (MaxwellKendall) [reply usaspending_api/references/v2/views/award_types.py]: I don't think so; we want the response to map at a more granular level than just `assistance` and `procurement`.

Here's the response object we want from the ticket itself:

```json
{
    "contracts": {
        "A": "BPA Call",
        "B": "Purchase Order",
        "C": "Delivery Order",
        "D": "Definitive Contract"
    },
    "loans": {
        "07": "Direct Loan",
        "08": "Guaranteed/Insured Loan"
    },
    "idvs": {
        "IDV_A": "GWAC Government Wide Acquisition Contract",
        "IDV_B": "IDC Multi-Agency Contract, Other Indefinite Delivery Contract",
        "IDV_B_A": "IDC Indefinite Delivery Contract / Requirements",
        "IDV_B_B": "IDC Indefinite Delivery Contract / Indefinite Quantity",
        "IDV_B_C": "IDC Indefinite Delivery Contract / Definite Quantity",
        "IDV_C": "FSS Federal Supply Schedule",
        "IDV_D": "BOA Basic Ordering Agreement",
        "IDV_E": "BPA Blanket Purchase Agreement"
    },
    "grants": {
        "02": "Block Grant",
        "03": "Formula Grant",
        "04": "Project Grant",
        "05": "Cooperative Agreement"
    },
    "other_financial_assistance": {
        "09": "Insurance",
        "11": "Other Financial Assistance"
    },
    "direct_payments": {
        "06": "Direct Payment for Specified Use",
        "10": "Direct Payment with Unrestricted Use"
    }
}
```

File: usaspending_api/api_contracts/contracts/v2/references/award_types.md (hunk 2 above)
Comment (MaxwellKendall) [reply usaspending_api/api_contracts/contracts/v2/references/award_types.md]: Pushed changes; thanks @tony-sappe

File: usaspending_api/api_contracts/contracts/v2/references/award_types.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    + Attributes (object)
+        + `contracts` (required, ContractTypes, fixed-type)
+        + `loans` (required, LoanTypes, fixed-type)
+        + `idvs` (required, IDVTypes, fixed-type)
+        + `grants` (required, GrantTypes, fixed-type)
+        + `other_financial_assistance` (required, OtherFinancialAssistanceTypes, fixed-type)
+        + `direct_payments` (required, DirectPaymentTypes, fixed-type)
+    + Body
+
+                {
```
Comment (sethstoudenmier) [inline usaspending_api/api_contracts/contracts/v2/references/award_types.md:19]: Can you correct the formatting of this body? Screenshot attached showing what it currently looks like: 
Screen Shot 2020-04-17 at 8 03 18 AM

Comment (MaxwellKendall) [reply usaspending_api/api_contracts/contracts/v2/references/award_types.md:19]: K

**Pull request merged**